src/managed/Microsoft.Extensions.DependencyModel/DependencyContextJsonReader.Netstandard2_0.cs

```diff
@@ -0,0 +1,628 @@
+            => tokenType == JsonTokenType.PropertyName;
+
+        private DependencyContext ReadCore(ReadOnlySpan<byte> jsonData)
+        {
+            var reader = new Utf8JsonReader(jsonData, isFinalBlock: true, state: default);
+
+            reader.ReadStartObject();
+
+            string runtime = string.Empty;
+            string framework = string.Empty;
+            bool isPortable = true;
+            string runtimeTargetName = null;
+            string runtimeSignature = null;
+
+            CompilationOptions compilationOptions = null;
+            List<Target> targets = null;
+            Dictionary<string, LibraryStub> libraryStubs = null;
+            List<RuntimeFallbacks> runtimeFallbacks = null;
+
+            while (reader.Read() && IsTokenTypeProperty(reader.TokenType))
+            {
+                switch (reader.GetString())
+                {
+                    case DependencyContextStrings.RuntimeTargetPropertyName:
+                        ReadRuntimeTarget(ref reader, out runtimeTargetName, out runtimeSignature);
+                        break;
+                    case DependencyContextStrings.CompilationOptionsPropertName:
+                        compilationOptions = ReadCompilationOptions(ref reader);
+                        break;
+                    case DependencyContextStrings.TargetsPropertyName:
+                        targets = ReadTargets(ref reader);
+                        break;
+                    case DependencyContextStrings.LibrariesPropertyName:
+                        libraryStubs = ReadLibraries(ref reader);
+                        break;
+                    case DependencyContextStrings.RuntimesPropertyName:
+                        runtimeFallbacks = ReadRuntimes(ref reader);
+                        break;
+                    default:
+                        reader.Skip();
+                        break;
+                }
+            }
+
+            if (compilationOptions == null)
+            {
+                compilationOptions = CompilationOptions.Default;
+            }
+
+            Target runtimeTarget = SelectRuntimeTarget(targets, runtimeTargetName);
+            runtimeTargetName = runtimeTarget?.Name;
+
+            if (runtimeTargetName != null)
+            {
+                var separatorIndex = runtimeTargetName.IndexOf(DependencyContextStrings.VersionSeparator);
+                if (separatorIndex > -1 && separatorIndex < runtimeTargetName.Length)
+                {
+                    runtime = runtimeTargetName.Substring(separatorIndex + 1);
+                    framework = runtimeTargetName.Substring(0, separatorIndex);
+                    isPortable = false;
+                }
+                else
+                {
+                    framework = runtimeTargetName;
+                }
+            }
+
+            Target compileTarget = null;
+
+            var ridlessTarget = targets.FirstOrDefault(t => !IsRuntimeTarget(t.Name));
```

Comment:
Nit: var => Target (I don't know what core-setup's coding guidelines are, but if it's the same as corefx, then please review the use of such var's, and I'll stop commenting on these)